test/test_distributed.py

```diff
@@ -168,6 +168,25 @@ def _init_global_test(self):
         rank = dist.get_rank()
         return (group, group_id, rank)
 
+    # HELPER FOR MULTIGPU TESTS
+    def _init_multigpu_helper(self):
+        """Multigpu tests are designed to simulate the multi nodes with multi
+        GPUs on each node. Nccl backend requires equal #GPUs in each process.
+        On a single node, all visible GPUs are evenly
+        divided to subsets, each process only uses a subset.
+        """
+        nGPUs = torch.cuda.device_count()
+        world_size = dist.get_world_size()
+        visible_devices = range(nGPUs)
+
+        if BACKEND == 'nccl':
+            apply_hack_for_nccl()
+
+        nGPUs_per_process = nGPUs // world_size
+        rank_to_GPU = {i: visible_devices[i * nGPUs_per_process: (i + 1) * nGPUs_per_process]
+                       for i in list(range(world_size))}
```

Comment:
That doesn't solve the issue, because the result of `visible_devices[...]` will be a `range` object. You need to do `rank_to_gpu = [list(visible_devices[...]) for i in range(world_size)]`